core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -3940,7 +3940,8 @@ public void doExit( StaplerRequest req, StaplerResponse rsp ) throws IOException
             w.println("Shutting down");
             w.close();
         }
-
+        // Just in case we have shutdown hooks
+        ACL.impersonate(ACL.SYSTEM);
```

Comment:
Pretty sure this won't do anything, shutdown hooks should run in a different thread.